File: ruby/vowel-count/vowel-count_spec.rb
require 'minitest/autorun'

module VowelCount
  extend self 

  def call(str)
    str.scan(/a|e|i|o|u/).size
  end
end

describe VowelCount do 
  it 'returns 0 for non vowels in a string' do 
    VowelCount.('pyx').must_equal 0
  end

  it 'returns 1 for one vocal' do 
    VowelCount.('a').must_equal 1
  end

  it 'returns 2 for two vocals' do 
    VowelCount.('ae').must_equal 2
  end

  it 'returns 3 for three vocals' do 
    VowelCount.('aei').must_equal 3
  end

  it 'returns 4 for four vocals' do 
    VowelCount.('aeio').must_equal 4
  end

  it 'returns 5 for five vocals' do 
    VowelCount.('aeiou').must_equal 5
  end
end

File: javascript/vowel-count/test/vowel-count_spec.js
var expect = require("chai").expect;

var getCount = function(str) { 
  var re = new RegExp("a|e|i|o|u", "g");
  var amount = str.match(re);
  return Array.isArray(amount) ? amount.length : 0;
}

describe("Vowel count", function() { 
  it("returns 0 for non vowel string", function() { 
    expect(getCount("pyx")).to.equal(0);
  });

  it("returns 1 for a", function() { 
    expect(getCount("a")).to.equal(1);
  })

  it("returns 2 for ae", function() { 
    expect(getCount("ae")).to.equal(2);
  })

  it("returns 3 for aei", function() { 
    expect(getCount("aei")).to.equal(3);
  })

  it("returns 4 for aeio", function() { 
    expect(getCount("aeio")).to.equal(4);
  })

  it("returns 5 for aeiou", function() { 
    expect(getCount("aeiou")).to.equal(5);
  })
})
